ADR/0004-component-image-location.md

```diff
@@ -0,0 +1,187 @@
+
+#### Lifecycle of the Quay.io resources
+
+* Token generation: 
+    * Robot account token: At the moment, the controller responsible for generating the Quay.io resources would be responsible for rotating the tokens https://docs.quay.io/api/swagger/#!/robot/regenerateUserRobotToken
+    * Quay.io API token: No programmatic way to regenerate this token is known at this point of time. This would be a manual activity to begin with. 
+
+* Upon deletion of an `Application`/`Component` from StoneSoup, 
+    * The controller/finalizer would delete the the relevant Quay.io resources namely, the image repository and the robot account.
+    * The controller/finalizer would delete the linked `Secret` from the user's namespace. Most likely, this should be a mere `ownerReference`-based garbage collection.
+
+* Upon removal of a user from Stonesoup,
+    * The empty Quay.io organization associated with the user or the user's Space may not be deleted instantly, but would be scheduled for a delayed cleanup.
+    
+    
+### How - Implementation
+
+The implementation of the above design will be improved overtime with the possible introduction of new CRDs/APIs. At the moment, no new API is being planned till the need for it arises.
+
+
+To request the Image controller to setup an image repository, annotate the `Component` with `image.redhat.com/generate: 'true'`.
+
+
+```
+apiVersion: StoneSoup.redhat.com/v1alpha1
+kind: Component
+metadata:
+  annotations:
+    image.redhat.com/generate: 'true'
+  name: billing
+  namespace: image-controller-system
+spec:
+  application: city-transit
+  componentName: billing
+```
+
+The `Image controller` creates the necessary resources on Quay.io and writes out the details of the same into the `Component` resource as an annotation, namely: 
+
+* The image repository URL.
+* The name of the Kubernets `Secret` in which the robot account token was written out to.
+
+```
+{
+   "image":"quay.io/redhat-user-workloads/image-controller-system/city-transit/billing",
+   "secret":"billing",
+}
+```
+
+```
+apiVersion: StoneSoup.redhat.com/v1alpha1
+kind: Component
+metadata:
+  annotations:
+    image.redhat.com/generate: 'false'
+    image.redhat.com/image: >-
+      {"image":"quay.io/redhat-user-workloads/image-controller-system/city-transit/billing","secret":"billing"
+      }
+  name: billing
+  namespace: image-controller-system
+  resourceVersion: '86424'
+  uid: 0e0f30b6-d77e-406f-bfdf-5802db1447a4
+spec:
+  application: city-transit
+  componentName: billing
+```
+    
+    
+
+
+## Open Questions
+
+- What would be a progammatic way to regenerate the main Quay.io API token ?
+- Since the long-term goal is to have the Quay.io organizations owned by the user, how do we build a frictionless experience to map the user's 
+  account with the user's Quay.io account ?
```

Comment:
Is there a way to transition between the types of quay ownership -- i.e. being able to hand over "keys" to an organization for an additional cost?

Reply:
That would be a customer support ticket I guess :-)